### DISCUSSION TITLE
Features: SMA in App Store

### DISCUSSION THREAD

**Discussion by Protektor-Desura** [IceWhaleTech/CasaOS#239] [Ideas]
Automatically convert video files to a standardized format with metadata tagging to create a beautiful and uniform media library using SMA. It is a plugin/script for Sonarr and Radarr. It allows to standardize your media and put metadata inside all your videos to make it easier to automatically identify them.

https://github.com/mdhiggins/sickbeard_mp4_automator
https://hub.docker.com/r/mdhiggins/sonarr-sma
https://hub.docker.com/r/mdhiggins/radarr-sma

**Comment by Lauren-ED209** [comment]
Wow. This is elegant!! I love this idea!

**Reply by Lauren-ED209** [reply]
Could you please share more about your scenarios, like your media player, self hosted app and general video downloading, transfer, management, and streaming process? I think it's really important for us to study the whole process before optimizing a standalone fragment.

**Comment by Protektor-Desura** [comment]
I suggest you look at a project I have been slowly working on that is similar to what CasaOS is trying to do. I have been working slowly on a complete home server project called Archon and have created a wiki for it full of all kinds of information about the design goals and why I chose what I chose.
https://github.com/Protektor-Desura/Archon

Archon is a little different because I also include a docker to stream video games to any device within the home. Basically cloud game streaming in the home so you don't need consoles on every TV or device in the home. Instead you stream your games from a master home server to save money and allow you to game from anywhere.

Everything I use is open-source with all the source code available under an open-source license and is no cost (no $$$) as well.


amd
I use AMD to automatically download music from Deezer using my account on Deezer.
https://github.com/RandomNinjaAtk/docker-amd

amtd
I use AMTD to download trailers and extras for my movies in Radarr.
https://github.com/RandomNinjaAtk/docker-amtd

amvd
I use AMVD to download music videos linking to artist from Lidarr.
https://github.com/RandomNinjaAtk/docker-amvd

arm
I use ARM to automatically rip all my existing physical media. All my music CDs, DVDs adn Bluray all automatically by just dropping the disc in the drive and let it process everything then it auto ejects the disc and I just drop the in the next disc without having to do anything and it read the disc then automatically pulls in metadata and stuffs it in the file.
https://github.com/automatic-ripping-machine/automatic-ripping-machine

bazarr
I use Bazarr to get subtitles for all my videos. It links to Radarr and Sonarr to find all your videos.
https://github.com/morpheus65535/bazarr

e2guardian
I use this to filter all of my web traffic and it has a MITM ability to even filter HTTPS connections as well. This lets me filter all ads at the URL level and filter other unwanted things for kids as well at the URL level instead of just at the DNS level.
https://github.com/e2guardian/e2guardian

embystat
I use embysat to get stats on Jellyfin that is similar to [Tautulli](https://tautulli.com/) some people use for Plex.
https://github.com/mregni/EmbyStat

ersatztv
I use this to create live TV channels using my own library of videos.
https://github.com/jasongdove/ErsatzTV

fhdhr
I use this to pull in free streaming live TV from the internet without needing cable TV or antenna device like HDHomeRun to get Live TV.
https://github.com/fHDHR/fHDHR

jdownloader2
I use this in a docker to download files (DDL) from the web to be imported in to Sonarr and Radarr.
https://github.com/svn2github/jdownloader2
https://svn.jdownloader.org/projects/jd

jellyfin
I use Jellyfin to serve up all my movies, tv shows, music, comic books, ebooks, audio books, podcasts, music videos. I use the OPDS plugin to support all OPDS ebook clients. It is whole discussion in of itself of all the Jellyfin plugins I use.
https://github.com/jellyfin/jellyfin

jellyfin-subsonic
I use this to support subsonic clients with Jellyfin.
https://github.com/nvllsvm/jellysub

lidarr
I use Lidarr to deal with my music and link it to beets to properly tag and get all music in the same format,
https://github.com/Lidarr/Lidarr
https://github.com/beetbox/beets

mylar3
I use Mylar3 to deal with all my comic books.
https://github.com/mylar3/mylar3

podgrab
I use this to find and pull in/download all the pod casts that I want.
https://github.com/akhilrex/podgrab

prowlarr
This is an indexer for torrents and usenet. It auto configures everything automatically for you with Lidarr, Sonarr, Radarr, Readarr, Mylar3.
https://github.com/Prowlarr/Prowlarr

qbittorrent
I use qBittorrent to pull in torrents. It has a web interface to monitor things plus there are other skins you can use with qBittorrent web interface as well.
https://github.com/qbittorrent/qBittorrent
https://github.com/qbittorrent/qBittorrent/wiki/List-of-known-alternate-WebUIs
https://flood.js.org/

radarr
I use Radarr to deal with my movies.
https://github.com/Radarr/Radarr
https://hub.docker.com/r/mdhiggins/radarr-sma

readarr
I use Readarr to deal with my ebook and audiobooks. The program currently can't do both at the same time so I run 2 copies, one for audiobooks and one for ebooks.
https://github.com/Readarr/Readarr

sonarr
I then use Sonarr to deal with my TV shows.
https://github.com/Sonarr/Sonarr
https://hub.docker.com/r/mdhiggins/sonarr-sma

unmanic
I use Unmanic with the SMA plugin to convert all my previously existing media to a standard setup to match SMA with Radarr and Sonarr.
https://github.com/Unmanic/unmanic



These are other programs outside of what is used to feed Jellyfin.

homeassistant
I use this to do all the local control of my home automation.
https://github.com/home-assistant/core

nextcloud
I use this to replace every cloud service out there like all the Google services and Dropbox/G-Drive, etc.
https://github.com/nextcloud/docker

portainer
I use this because it has more features currently for managing dockers from the web than CasaOS currently does.
https://github.com/portainer/portainer

watchtower
I use this to keep all my dockers up to date.
https://github.com/containrrr/watchtower

**Reply by Lauren-ED209** [reply]
Will study carefully what you share. Won't waste your time. Thanks!
